src/demo-app/a11y/grid-list/grid-list-a11y.html

```diff
@@ -0,0 +1,42 @@
+<section>
+  <h2>Fixed-height grid list</h2>
+  <md-grid-list [cols]="fixedCols" [rowHeight]="fixedRowHeight">
+    <md-grid-tile *ngFor="let tile of tiles" [colspan]="tile.cols" [rowspan]="tile.rows"
+                  [style.background]="tile.color">
+      {{tile.text}}
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Ratio-height grid list</h2>
+  <md-grid-list cols="2" [rowHeight]="ratio" gutterSize="4px">
+    <md-grid-tile *ngFor="let tile of tiles" [style.background]="'lightblue'">
+      {{tile.text}}
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Fit-height grid list</h2>
+  <md-grid-list cols="2" rowHeight="fit" [gutterSize]="ratioGutter"
+                [style.height]="fitListHeight">
+    <md-grid-tile *ngFor="let tile of tiles" [style.background]="'#F1EBBA'">
+      {{tile.text}}
+    </md-grid-tile>
+  </md-grid-list>
+</section>
+
+<section>
+  <h2>Grid list with header and footer</h2>
+  <md-grid-list cols="3" rowHeight="200px">
+    <md-grid-tile *ngFor="let dog of dogs">
+      <md-grid-tile-header>{{dog.name}}</md-grid-tile-header>
+      <img [alt]="dog.name" src="https://material.angularjs.org/material2_assets/ngconf/{{dog.name}}.png">
```

Comment:
I think `role="listitem"` is okay to contain images. The description in w3c https://www.w3.org/TR/wai-aria/roles#list says it' document structure with non-interactive elements. It doesn't say it cannot contain images. 

I'm okay to remove the default roles in grid list and grid tile, as users may want to use them as interactive elements.